lib/share_progress/button.rb

```diff
@@ -1,27 +1,26 @@
 require 'share_progress'
 require 'share_progress/client'
+require 'share_progress/utils'
 
 module ShareProgress
   class Button
 
-    attr_accessor :page_url, :page_title, :button_template, :share_button_html, :is_active
-    attr_reader   :id
+    attr_accessor :page_url, :page_title, :button_template, :share_button_html, :is_active, :auto_fill, :variations, :advanced_options
+    attr_reader   :id, :errors
 
     class << self
 
-      def create(page_url, button_template, raw_options={})
-        created = update(nil, page_url, button_template, raw_options)
-        created.nil? ? nil : new(created)
+      def create(options={})
+        created = update(options)
```

Comment:
hey @osahyoun do you think that I should raise ArgumentError if options doesn't include the two parameters required by the API? or should I let that error trickle down into the validation errors on the button instance (as it does now)